CardanoSharp.Wallet.Test/HDWallet/DerivationTests.cs

```diff
@@ -0,0 +1,204 @@
+                                .Derive(PurposeType.Shelley)    // IPurposeNodeDerivation
+                                .Derive(CoinType.Ada)           // ICoinNodeDerivation
+                                .Derive(0)                      // IAccountNodeDerivation
+                                .Derive(RoleType.ExternalChain) // IRoleNodeDerivation
+                                .Derive(0);                     // IIndexNodeDerivation
+
+            // Assert
+            Assert.Equal(paymentPrv1.Key, derivation.PrivateKey.Key);
+            Assert.Equal(paymentPrv1.Chaincode, derivation.PrivateKey.Chaincode);
+
+            Assert.Equal(paymentPub1.Key, derivation.PublicKey.Key);
+            Assert.Equal(paymentPub1.Chaincode, derivation.PublicKey.Chaincode);
+        }
+
+
+        [Theory]
+        [InlineData(__mnemonic, "m/1852'")]
+        [InlineData(__mnemonic, "m/1852'/1815'")]
+        [InlineData(__mnemonic, "m/1852'/1815'/0'")]
+        [InlineData(__mnemonic, "m/1852'/1815'/0'/0")]
+        [InlineData(__mnemonic, "m/1852'/1815'/0'/0/0")]
+        public void PartialPathDerivationTest(string words, string path)
+        {
+            var depth = path.Split("/").Slice(1).Count(); // strip master node to calc zero based depth
+
+            // create two payment addresses from same root key
+            //arrange
+            var mnemonic = new KeyService().Restore(words);
+            var rootKey = mnemonic.GetRootKey();
+            var testKey = getTestRootKey(mnemonic);
+
+            (var prv, var pub) = getKeyPairFromPath(path, rootKey);
+
+            // Act
+            // Fluent derivation API
+            var master = testKey.Derive();
+
+            var purpose = master.Derive(PurposeType.Shelley);
+            var coin = purpose.Derive(CoinType.Ada);
+            var account = coin.Derive(0);
+            var role = account.Derive(RoleType.ExternalChain);
+            var index = role.Derive(0);
+
+            if (depth == 1) AssertDerivedKeys(prv, pub, purpose);
+            if (depth == 2) AssertDerivedKeys(prv, pub, coin);
+            if (depth == 3) AssertDerivedKeys(prv, pub, account);
+            if (depth == 4) AssertDerivedKeys(prv, pub, role);
+            if (depth == 5) AssertDerivedKeys(prv, pub, index);
+        }
+
+        [Fact]
+        public void ImplicitPathDerivationTest()
+        {
+            var path = "m/1852'/1815'/0'/0/0";
+
+            // Arrange
+            var mnemonic = new KeyService().Restore(__mnemonic);
+            var rootKey = mnemonic.GetRootKey();
+            var testKey = getTestRootKey(mnemonic);
+
+            (var paymentPrv1, var paymentPub1) = getKeyPairFromPath(path, rootKey);
+
+            // Act
+            // Fluent derivation API
+            var derivation = testKey.Derive(RoleType.ExternalChain)      // implicit IRoleNodeDerivation
+                                    .Derive(0);                          // IIndexNodeDerivation
```

Comment:
starts *on* account (implicit: `1852'` / `1815'`)
then derives 0' (explicitly)

Reply:
eh, sort of. 

Use Case:
I'm writing a light client (similar to Yoroi, Nami, C64Minter, etc). User enters in the Mnemonic and Spending Password. I derive down to the Account level and get the Public Key. I use the Spending Password to 2-Way encrypt the Private Key and then store both the encrypted Private Key and the plain Public Key. 

The user wants to generate a random/incremented address. I take the Public Key and Derive (0/1). 
```csharp
var paymentKey = accountPublicKey.Derive("0/1");
```
so I have the Account level key and want to continue the derivation process.

The user now wants to send a transaction. They enter in the ADDR, Amount to Send, and their Spending Password. I can now decrypt the Account Private Key and derive down to the "0/1" Index Private Key. This allows me to now sign the Transaction. 

So in both cases, I'm not starting with the root. I'm starting with the Account Keys. I don't know nor need the `m/1852/1815` keys. This only works because `Role` and `Index` are soft derivations.